packages/builder/src/stores/portal/plugins.ts

```diff
@@ -1,17 +1,13 @@
 import { writable } from "svelte/store"
 import { PluginSource } from "constants/index"
-
+import { Plugin } from "@budibase/types"
 import { API } from "api"
 
-interface Plugin {
-  _id: string
-}
-
 export function createPluginsStore() {
   const { subscribe, set, update } = writable<Plugin[]>([])
 
   async function load() {
-    const plugins = await API.getPlugins()
+    const plugins = (await API.getPlugins()) as Plugin[]
```

Comment:
Could the API not be typed, instead?

Reply:
The API actually is typed and returns a `FetchPluginResponse`, which is typed as
```js
export type FetchPluginResponse = Plugin[]
```
so I'm only casting to `Plugin[]` because I'd prefer to store that type, even though there is no difference. It just feels wrong to store `FetchPluginResonse` in the store when it's actually just an array of plugins. I'm happy remove this if you'd prefer - I'd actually appreciate knowing what you think is better because I knew this casting was redundant but it just seemed "nicer".

Reply:
I think my preference is to do something like `const plugins: Plugin[] = await API.getPlugins()` in these type of scenarios, this isn't casting in this case as Typescript will infer whether the types overlap enough for it to work.

Using `as Plugin[]` is more dangerous, because you are forcing the conversion - there is some protections still if the two types don't overlap at all/it makes no sense, but you'll lose protections around things like undefined checks.

Reply:
Sounds good! I'll go for that approach instead of the `as xxx`. Thank you 🙏